File: 0901-online-stock-span/0901-online-stock-span.py
class StockSpanner:

    def __init__(self):
        self.st=[]
        self.ind=-1

    def next(self, price: int) -> int:
        self.ind+=1
        while(self.st and self.st[-1][0]<=price):
            self.st.pop()
        ans=self.ind-(self.st[-1][1] if self.st else-1)
        self.st.append((price,self.ind))
        return ans


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: 0901-online-stock-span/0901-online-stock-span.java
class StockSpanner {

    private Deque<int[]> st;
    private int ind;
    public StockSpanner() {
        st=new ArrayDeque<>();
        ind=-1;
    }
    
    public int next(int price) {
        ind=ind+1;
        while(!st.isEmpty() && st.peek()[0]<=price) st.pop();
        int ans=ind-(st.isEmpty()?-1:st.peek()[1]);
        st.push(new int[]{price,ind});
        return ans;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */